### ISSUE TITLE
Pull/Pull Not Works - Not Works With Publickey

### ISSUE THREAD

**Issue by wilkis3** [WeblateOrg/weblate#9016]
### Describe the issue

Push and pull with SSH on selfhosted Weblate and selfhosted GitLab instance did not worked.

Works good on GitHub before.

The translation user have Owner permissions in the repository now. GitLab shows that the Weblate SSH key was never used.

### I already tried

- [X] I've read and searched [the documentation](https://docs.weblate.org/).
- [X] I've searched for similar issues in this repository.

### Steps to reproduce the behavior

**Pull**
1. Go to the component settings
2. Enter git SSH URL e.g. `git@git.example:code/repo.git` in Source code repository field
3. Click on save

**Push**
1. Go to the component settings
2. Enter git SSH URL e.g. `git@git.example:code/repo.git` in Repository push URL  field
3. Click save
4. Go to the repository maintenance at the some component
5. Click on Push
6. Get the error message banner " Could not push to remote branch on *component name*"

### Exception traceback

```pytb
Docker logs:


gunicorn stderr | [2023-03-30 11:57:23,939: INFO/374] projectname/componentname-json: updating repository
gunicorn stderr | [2023-03-30 11:57:24,870: INFO/374] projectname/componentname-json: update took 0.93 seconds
gunicorn stderr | [2023-03-30 11:57:24,870: DEBUG/374] projectname/componentname-json: update: warning: redirecting to https://git.example/code/projectname.git/
gunicorn stderr | [2023-03-30 11:57:24,876: INFO/374] projectname/componentname-json: repository up to date at cb5748b3493a65eda0ce9a5bd6a7991faa6de9d2
gunicorn stderr | [2023-03-30 11:57:24,918: INFO/374] projectname/componentname-json: pushing to remote repo
2023-03-30 11:57:35,797 INFO reaped unknown pid 1055 (exit status 255)
gunicorn stderr | [2023-03-30 11:57:35,796: DEBUG/374] git: failure git@git.example: Permission denied (publickey).
gunicorn stderr | fatal: Could not read from remote repository.
gunicorn stderr |
gunicorn stderr | Please make sure you have the correct access rights
gunicorn stderr | and the repository exists.
gunicorn stderr |  (128)
gunicorn stderr | [2023-03-30 11:57:35,796: DEBUG/374] git: failure git@git.example: Permission denied (publickey).
gunicorn stderr | fatal: Could not read from remote repository.
gunicorn stderr |
gunicorn stderr | Please make sure you have the correct access rights
gunicorn stderr | and the repository exists.
gunicorn stderr |  (128)
gunicorn stderr | [2023-03-30 11:57:35,814: DEBUG/374] git: On branch develop
gunicorn stderr | Your branch is ahead of 'origin/develop' by 1 commit.
gunicorn stderr |   (use "git push" to publish your local commits)
gunicorn stderr |
gunicorn stderr | nothing to commit, working tree clean
gunicorn stderr |
gunicorn stderr | [2023-03-30 11:57:35,814: DEBUG/374] git: On branch develop
gunicorn stderr | Your branch is ahead of 'origin/develop' by 1 commit.
gunicorn stderr |   (use "git push" to publish your local commits)
gunicorn stderr |
gunicorn stderr | nothing to commit, working tree clean
gunicorn stderr |
gunicorn stderr | [2023-03-30 11:57:35,827: WARNING/374] Could not push the repo: RepositoryException: git@git.example: Permission denied (publickey).
gunicorn stderr | fatal: Could not read from remote repository.
gunicorn stderr |
gunicorn stderr | Please make sure you have the correct access rights
gunicorn stderr | and the repository exists.
gunicorn stderr |  (128)
celery-notify stderr | [2023-03-30 11:57:35,846: INFO/MainProcess] Task weblate.accounts.tasks.notify_change[1427a70c-c0b5-420d-910f-e561c08283cb] received
```

Sentry shows not really more informations.

### How do you run Weblate?

Docker container

### Weblate versions

 * Weblate: 4.16.4
 * Django: 4.1.7
 * siphashc: 2.1
 * translate-toolkit: 3.8.6
 * lxml: 4.9.2
 * Pillow: 9.4.0
 * nh3: 0.2.8
 * python-dateutil: 2.8.2
 * social-auth-core: 4.4.0
 * social-auth-app-django: 5.1.0
 * django-crispy-forms: 2.0
 * oauthlib: 3.2.2
 * django-compressor: 4.3.1
 * djangorestframework: 3.14.0
 * django-filter: 22.1
 * django-appconf: 1.0.5
 * user-agents: 2.2.0
 * filelock: 3.10.0
 * rapidfuzz: 2.13.7
 * openpyxl: 3.1.2
 * celery: 5.2.7
 * django-celery-beat: 2.5.0
 * kombu: 5.2.4
 * translation-finder: 2.15
 * weblate-language-data: 2023.3
 * html2text: 2020.1.16
 * pycairo: 1.23.0
 * pygobject: 3.42.2
 * diff-match-patch: 20200713
 * requests: 2.28.2
 * django-redis: 5.2.0
 * hiredis: 2.2.2
 * sentry_sdk: 1.16.0
 * Cython: 0.29.33
 * misaka: 2.1.1
 * GitPython: 3.1.31
 * borgbackup: 1.2.3
 * pyparsing: 3.0.9
 * pyahocorasick: 2.0.0
 * python-redis-lock: 4.0.0
 * charset-normalizer: 3.1.0
 * Python: 3.11.2
 * Git: 2.30.2
 * psycopg2: 2.9.5
 * psycopg2-binary: 2.9.5
 * phply: 1.2.6
 * ruamel.yaml: 0.17.21
 * tesserocr: 2.6.0
 * boto3: 1.26.92
 * zeep: 4.2.1
 * aeidon: 1.12
 * iniparse: 0.5
 * mysqlclient: 2.1.1
 * Mercurial: 6.3.3
 * git-svn: 2.30.2
 * git-review: 2.3.1
 * Redis server: 6.2.11
 * PostgreSQL server: 14.7
 * Database backends: django.db.backends.postgresql
 * Cache backends: default:RedisCache, avatar:FileBasedCache
 * Email setup: django.core.mail.backends.smtp.EmailBackend: mail.your-server.de
 * OS encoding: filesystem=utf-8, default=utf-8
 * Celery: redis://cache:6379/1, redis://cache:6379/1, regular
 * Platform: Linux 5.4.0-146-generic (x86_64)

### Weblate deploy checks

```shell
System check identified some issues:

WARNINGS:
?: (security.W018) You should not have DEBUG set to True in deployment.

INFOS:
?: (weblate.I028) Backups are not configured, it is highly recommended for production use
	HINT: https://docs.weblate.org/en/weblate-4.16.4/admin/backup.html
```

**Comment by wilkis3** [comment]
> Please verify that the public key imported to GitLab matches the Weblate one. (It is available at `/keys/` URL.)

Both public keys are exactly the same.

**Comment by wilkis3** [comment]
> > Please verify that the public key imported to GitLab matches the Weblate one. (It is available at `/keys/` URL.)
> 
> Both public keys are exactly the same.

And the private key matchs to the public key. (Same SSH RSA fingerprint) . RSA 4069 Key are also allowed at this GitLab instance.

**Comment by wilkis3** [comment]
Thank you @nijet 

The SSH connection was blocked by the SSHd of the GitLab instance. RSA key are not allowed there any more.

**Comment by nijel** [comment]
Good point, I've implemented support for ed25519 in https://github.com/WeblateOrg/weblate/pull/9025
